### PR TITLE
Fixed Syntax and Type errors

### PR DESCRIPTION
**Pull request by AmritKumar**
This fixes the syntax error reported in https://github.com/Xfers/XSGD-scilla/issues/8

### CODE REVIEW

File: contracts/xsgd_contract.scilla [hunk 1]
```diff
@@ -370,11 +370,10 @@ transition lawEnforcementWipingBurn(address : ByStr20, initiator : ByStr20)
       | Some b =>
         balances[address] := zero;
         currentSupply <- totalSupply;
-        new_supply = builtin sub currentSupply bal;
+        new_supply = builtin sub currentSupply b;
```
Comment (junhaoyap) [inline contracts/xsgd_contract.scilla]: @AmritKumar this is fixed in https://github.com/Xfers/XSGD-scilla/pull/13/files so I'm removing this from this PR

Comment (junhaoyap) [reply contracts/xsgd_contract.scilla]:
```suggestion
        new_supply = builtin sub currentSupply bal;
```

File: contracts/wallet.scilla [hunk 2]
```diff
@@ -207,13 +207,13 @@ let custom_token_transaction_msg_as_list =
         {_recipient : proxyTokenContract ;
          _tag : "proxyPause" ;
          _amount : Uint128 0 }
-      | Burn =>
+      | Burn value =>
```
Comment (junhaoyap) [inline contracts/wallet.scilla:210]: Looks good this is the exact same line I had in mind

File: contracts/wallet.scilla [hunk 3]
```diff
@@ -207,13 +207,13 @@ let custom_token_transaction_msg_as_list =
...
         (* Burn             (value : Uint128)  *)
         {_recipient : proxyTokenContract ;
          _tag : "proxyBurn" ;
          _amount : Uint128 0 ;
          value : value }
-      | Mint =>
+      | Mint to value =>
```
Comment (junhaoyap) [inline contracts/wallet.scilla:216]: Looks good this is the exact same line I had in mind

**Review by jjcnn — APPROVED** [review]

**Pull request merged**